node/network/approval-distribution/src/lib.rs

```diff
@@ -719,8 +1007,8 @@ impl State {
 		ctx: &mut Context,
 		metrics: &Metrics,
 		source: MessageSource,
-		assignment: IndirectAssignmentCert,
-		claimed_candidate_index: CandidateIndex,
+		assignment: IndirectAssignmentCertV2,
```

Comment:
likewise here, if this a `RelayVrfModuloCompact` - the `CoreBitfield` size needs to be checked as one of the first actions.

Reply:
and the `CoreBitfield` should have the same length as the `CandidateBitfield`

Reply:
> and the `CoreBitfield` should have the same length as the `CandidateBitfield`

Yes this should be done in `approval distribution` as proposed by https://github.com/paritytech/polkadot/pull/6782#discussion_r1277563323

Reply:
done